crypto/x509/x509_vfy.c

```diff
@@ -267,9 +267,12 @@ int X509_verify_cert(X509_STORE_CTX *ctx)
     int ret;
 
     if (ctx->cert == NULL) {
-        ERR_raise(ERR_LIB_X509, X509_R_NO_CERT_SET_FOR_US_TO_VERIFY);
-        ctx->error = X509_V_ERR_INVALID_CALL;
-        return -1;
+        if (sk_X509_num(ctx->untrusted) < 1) {
+            ERR_raise(ERR_LIB_X509, X509_R_NO_CERT_SET_FOR_US_TO_VERIFY);
+            ctx->error = X509_V_ERR_INVALID_CALL;
+            return -1;
+        }
+        ctx->cert = sk_X509_value(ctx->untrusted, 0);
```

Comment:
Though passing no cert and no untrusted certs is invalid call, I agree that checking for this rather than crashing is reasonable. Thanks!

Reply:
My pleasure.
I've meanwhile improved the addition by adding two related safety checks
and made use of the new feature to simplify `test/danetest.c`.
Interestingly, there were no other uses within OpenSSL that could take (direct) advantage of it.